src/main/kotlin/no/nav/familie/ef/iverksett/brev/BrevFunctions.kt

```diff
@@ -33,13 +35,22 @@ fun lagStønadtypeTekst(stønadstype: StønadType): String =
             StønadType.SKOLEPENGER -> "stønad til skolepenger"
         }
 
-fun lagVedtakstekst(vedtaksresultat: Vedtaksresultat, behandlingÅrsak: BehandlingÅrsak): String =
-    if (behandlingÅrsak == BehandlingÅrsak.SANKSJON_1_MND) {
-        "Vedtak om sanksjon av "
-    } else {
-        when (vedtaksresultat) {
-            Vedtaksresultat.INNVILGET -> "Vedtak om innvilgelse av "
-            Vedtaksresultat.AVSLÅTT -> "Vedtak om avslag av "
-            Vedtaksresultat.OPPHØRT -> "Vedtak om opphør av "
-        }
-    }
\ No newline at end of file
+fun lagVedtakstekst(iverksett: Iverksett): String =
+        if (iverksett.behandling.behandlingType === BehandlingType.FØRSTEGANGSBEHANDLING){
+            when (iverksett.vedtak.vedtaksresultat) {
+                    Vedtaksresultat.INNVILGET -> "Vedtak om innvilget "
+                    Vedtaksresultat.AVSLÅTT -> "Vedtak om avslått "
+                    Vedtaksresultat.OPPHØRT -> "Vedtak om opphørt "
+                }
+        } else {
+            if(iverksett.behandling.behandlingÅrsak === BehandlingÅrsak.SANKSJON_1_MND){
+                "Vedtak om sanksjon av "
+            }
+            else {
+                if(iverksett.vedtak.vedtaksresultat === Vedtaksresultat.OPPHØRT){
+                    "Vedtak om opphørt "
+                } else {
+                    "Vedtak om revurdert "
+                }
+            }
+        }
```

Comment:
Tar gjerne innspill på hvordan denne kan løses bedre

Reply:
Men her har vi ikke håndtering av avslått revurdering? 
Vi har opphørt på førstegangsbehandling, som ikke er mulig?

Reply:
En "avslått revurdering" blir "Vedtak om revurdert ...", men man kan jo diskutere om det er riktig.

Opphørt førstegangsbehandling gir heller ikke mening, så da kan kanskje kaste feil for å unngå forvirring.

Reply:
Liker en litt komprimert versjon med `when`